src/lib/builders/menu/create.ts

```diff
@@ -1083,49 +1083,52 @@ export function createMenuBuilder(opts: _MenuBuilderOptions) {
 		}
 	});
 
-	effect([rootOpen, rootActiveTrigger], ([$rootOpen, $rootActiveTrigger]) => {
-		if (!isBrowser) return;
-
-		const unsubs: Array<() => void> = [];
+	effect(
+		[rootOpen, rootActiveTrigger, preventScroll],
+		([$rootOpen, $rootActiveTrigger, $preventScroll]) => {
+			if (!isBrowser) return;
 
-		if ($rootOpen && get(preventScroll)) {
-			unsubs.push(removeScroll());
-		}
+			const unsubs: Array<() => void> = [];
 
-		if (!$rootOpen && $rootActiveTrigger) {
-			handleRovingFocus($rootActiveTrigger);
-		}
+			if (opts.selector !== 'menubar-menu' && $rootOpen && $preventScroll) {
+				unsubs.push(removeScroll());
+			}
```

Comment:
Let's add a comment here explaining the menubar scroll is handled on their end. Alternatively, this could be an option of createMenu?